src/core/openthread-core-config.h

```diff
@@ -44,14 +44,16 @@
 
 #define OPENTHREAD_THREAD_VERSION (2)
 
+// This should be before openthread-core-default-config.h
+#include "config/coap.h"
```

Comment:
One suggestion here: How about we include the header file from the file that needs/uses the other definitions
For example, `openthread-core-default-config.h` wants to use  `OPENTHREAD_CONFIG_COAP_SECURE_API_ENABLE`, so we add `#include "config/coap.h"` at the top of it.

Same way for `config/time_sync` and `config/mac`, i.e., if `config/mac.h` needs/uses `OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT`, we can add `#include "config/time_sync.h"` in `config/mac.h`.

The advantage I see with this approach is that the dependencies are handled by the header files themselves (instead of order of includes in this file), and if we ever happen  to have a loop of header file dependencies we would notice it.

Thoughts?

Reply:
Great, I agree.